### DISCUSSION TITLE
lang Extras auto-install LSP not working

### DISCUSSION THREAD

**Discussion by ben-haas** [LazyVim/LazyVim#5197] [Q&A] [answered]
I've enabled some languages in the LazyExtras, but the LSP's and some other items don't seem to be installing automatically in Mason. For example, in the docs for the Go extra, `goimports`, `gofumpt`, `gomodifytags`, `impl`, and `delve` should be added to the Mason `ensure_installed` option, however only `goimports` and `gofumpt` are being installed automatically. The `gopls` LSP is also not being installed automatically.

I'm using **v14.5.0**

![image](https://github.com/user-attachments/assets/e3053c77-bab7-4593-bcf6-acd2f612bce0)

![image](https://github.com/user-attachments/assets/f60c72a2-402c-44c3-be73-c61120e1bb56)

![image](https://github.com/user-attachments/assets/3021d25b-4d86-4ddb-a31d-54d252905b75)

**Comment by dpetka2001** [comment, marked as answer]
I just answered yesterday specifically [here](https://github.com/LazyVim/LazyVim/issues/5173#issuecomment-2550984952) also showing a screencast of installing the Go Extra. `gopls` got installed correctly and I just tried now again with a fresh LazyVim installation. Perhaps you're overriding the `config` function from `nvim-lspconfig` in your personal configuration?

**Reply by ben-haas** [reply]
Aha! I am using the config to add sourcekit-lsp. This is the only way I've been able to get it to work. I commented out the config and everything loaded up correctly.

![image](https://github.com/user-attachments/assets/6d310dfd-3380-4aa6-aa0d-f6a43c920a9c)

**Reply by dpetka2001** [reply]
You could try doing
```lua
"neovim/nvim-lspconfig",
opts = {
  servers = {
    sourcekit = {
      capabilities = .....  --rest of your stuff here
    },
  },
},
```

**Reply by ben-haas** [reply]
That works! Thank you. I'm still pretty new to LazyVim.

**Reply by dpetka2001** [reply]
Please try avoiding using the `config` function in Neovim distros in general. Usually some stuff are configured there and if you also use `config` in your personal configuration you will overwrite them. Try using `opts` instead, because everything you customize in `opts` will get merged instead of overwritten. Also, I highly recommend reading the LazyVim [docs](https://www.lazyvim.org) and ask questions if something is not clear. 

I'm just a simple user like you, but I will try to reply to the best of my knowledge (because there are many things that I myself don't understand that well).
